pydrive2/fs/spec.py

```diff
@@ -534,13 +534,22 @@ def _gdrive_upload_fobj(self, title, parent_id, stream, callback=None):
         return item
 
     def cp_file(self, lpath, rpath, **kwargs):
-        """In-memory streamed copy"""
-        with self.open(lpath) as stream:
-            # IterStream objects doesn't support full-length
-            # seek() calls, so we have to wrap the data with
-            # an external buffer.
-            buffer = io.BytesIO(stream.read())
-            self.upload_fobj(buffer, rpath)
+        """copy a single file or folder"""
+
+        if self.exists(rpath):
+            raise FileExistsError(rpath)
+
+        file_id = self._get_item_id(lpath)
```

Comment:
how about we use `info` here? to get all the information at once .. (info itself looks like can be optimized - it takes 2 API calls now, should be one)

Reply:
later i need the file as `GoogleDriveFile`.
also `info` will fetch all the metadata anyway.

Reply:
that's what I mean, ideally we should be doing one API call to get id _and_ metadata

Reply:
`info` does not return the id:

https://github.com/simone-viozzi/PyDrive2/blob/e765b584ff862e838c794823bd037e5f1bbf62f6/pydrive2/fs/spec.py#L409-L425